### ISSUE TITLE
Missing failover support when connecting to a cluster

### ISSUE THREAD

**Issue by ebuildy** [phpredis/phpredis#1400]
Relate to https://github.com/phpredis/phpredis/issues/1222

### Expected behaviour

failover must work when retrieving cluster slots at initialization.

### Actual behaviour

On a cluster environment, at initialization, we must retrieve the ``CLUSTER SLOT``, if the socket connection failed, an exception is thrown and failover is not working here, here the call stack:

1. https://github.com/phpredis/phpredis/blob/develop/cluster_library.c#L615
2. https://github.com/phpredis/phpredis/blob/1d997873750e5fdef1fe368b5296f4427dab3ea0/library.c#L2064

### I'm seeing this behaviour on
- OS: Linux Debian
- Redis: 3.x
- PHP: 5.6
- phpredis: 4.1.1

### I've checked
- [X] There is no similar issue from other users
- [] Issue isn't fixed in `develop` branch

**Comment by CZechBoY** [comment]
btw [php5.6 & php7.0 is dead](http://php.net/supported-versions.php), use php7.1 or php7.2

**Comment by ebuildy** [comment]
Not sure this is related, but thanks you...

**Comment by Gared** [comment]
I was wondering if this functionality was working in the past anytime, because in the cluster readme you can read this: ```...we have to re-issue the command to a different node (either because Redis cluster responded with MOVED/ASK or because we failed to communicate with a given node).```
So my understanding is that if the first socket connection to a node fails it will try to connect to other nodes until the timeout is reached, but (sadly) it never worked for me that way.
Did I misunderstood something?

**Comment by BrandtM** [comment]
@Gared @ebuildy I could reproduce this problem as well. Whenever the first master it tried to connect to was down the connection just didn't work. 
I fixed this issue because we have this issue in our company as well and I was tasked with fixing it. I looked into alternatives and it seems that predis could deal with it but had some drawbacks and we prefer phpredis.  
I looked into other ways of fixing it (i.e. adding an optional parameter) but the solution I chose, which is to unconditionally apply this fix in the RedisCluster constructor, was the least intrusive that didn't require other devs to rewrite their code.
